core/schema_utils.py

```diff
@@ -544,11 +544,10 @@ def is_valid_math_expression(obj: str, algebraic: bool) -> bool:
         if not expression_parser.is_valid_expression(obj): # type: ignore[no-untyped-call]
             return False
 
-        expression_is_algebraic = expression_parser.is_algebraic(obj) # type: ignore[no-untyped-call]
-        # If the algebraic flag is true, expression_is_algebraic should
-        # also be true, otherwise both should be false which would imply
-        # that the expression is numeric.
-        return not algebraic ^ expression_is_algebraic
+        expression_contains_at_least_one_variable = (
+            expression_parser.contains_at_least_one_variable(obj)) # type: ignore[no-untyped-call]
+        # This ensures that numeric expressions don't contain variables.
+        return algebraic or not expression_contains_at_least_one_variable
```

Comment:
Do we have test coverage for the new checks here?

Reply:
I've updated the tests for the new checks.